server/src/test/java/keywhiz/inject/DataSourceModuleTest.java

```diff
@@ -0,0 +1,31 @@
+package keywhiz.inject;
+
+import com.google.inject.Guice;
+import io.dropwizard.db.ManagedDataSource;
+import javax.inject.Inject;
+import keywhiz.service.config.Readonly;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import static org.junit.Assert.assertSame;
+import static org.mockito.Mockito.mock;
+
+public class DataSourceModuleTest {
+  @Test
+  public void injectsDataSources() {
+    class Holder {
+      @Inject ManagedDataSource readWrite;
+      @Inject @Readonly ManagedDataSource readonly;
+    }
+
+    Holder holder = new Holder();
```

Comment:
Anonymous inner classes (one of Java's lesser-known features) are a fantastic way to test dependency injection in a convincing way: you can declare fields with various annotations exactly as you would in your regular classes.

Reply:
TIL!